src/include/function/aggregate/collect.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "common/in_mem_overflow_buffer_utils.h"
+#include "common/vector/value_vector_utils.h"
+#include "processor/result/factorized_table.h"
+
+namespace kuzu {
+namespace function {
+
+struct CollectFunction {
+
+    struct CollectState : public AggregateState {
+        CollectState() : factorizedTable{nullptr} {}
+        inline uint32_t getStateSize() const override { return sizeof(*this); }
+        void moveResultToVector(common::ValueVector* outputVector, uint64_t pos) override {
+            auto dstKUList = outputVector->getValue<common::ku_list_t>(pos);
+            auto numBytesPerElement =
+                factorizedTable->getTableSchema()->getColumn(0)->getNumBytes();
+            dstKUList.size = factorizedTable->getNumTuples();
+            dstKUList.overflowPtr =
+                reinterpret_cast<uint64_t>(outputVector->getOverflowBuffer().allocateSpace(
+                    factorizedTable->getNumTuples() * numBytesPerElement));
+            outputVector->setValue<common::ku_list_t>(pos, dstKUList);
+            switch (outputVector->dataType.childType->typeID) {
+            case common::STRING: {
+                for (auto i = 0u; i < dstKUList.size; i++) {
+                    common::InMemOverflowBufferUtils::copyString(
+                        *reinterpret_cast<common::ku_string_t*>(factorizedTable->getTuple(i)),
+                        reinterpret_cast<common::ku_string_t*>(dstKUList.overflowPtr)[i],
+                        outputVector->getOverflowBuffer());
+                }
+            } break;
+            case common::LIST: {
+                for (auto i = 0u; i < dstKUList.size; i++) {
+                    common::InMemOverflowBufferUtils::copyListRecursiveIfNested(
+                        *reinterpret_cast<common::ku_list_t*>(factorizedTable->getTuple(i)),
+                        reinterpret_cast<common::ku_list_t*>(dstKUList.overflowPtr)[i],
+                        *outputVector->dataType.childType, outputVector->getOverflowBuffer());
+                }
+            } break;
+            default: {
+                for (auto i = 0u; i < dstKUList.size; i++) {
```

Comment:
Can we do bulk copy here? Instead of copy one value at a time.
You will need to access data blocks in fTable, and try to copy one block at a time.

Reply:
No, we store the null bits at the end of each tuple, we shouldn't copy the null bits.